### PR TITLE
Add nginx-php service

### LINKED ISSUE #18: Create service for multiple PHP versions on nginx
Create a service that will allow for development in PHP with an nginx web server.

Requirements:

- Should be able to run several PHP instances at the same time
- Each version of PHP should be able to have a build script to allow for installing of PHP modules
- Should have a configurable webroot directory that the user can point to that could exist outside of the workspace directory
- Proper documentation so that the user is able to understand and use the service.

### PR DESCRIPTION
**Pull request by joshpennington**
Resolves #18

This adds a container that is useful for running an nginx server will many versions of PHP available to it. This container is more complex than other so I have added a Readme file to explain how to get this setup.

### CODE REVIEW

File: services/nginxphp/php54/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM php:5.4-fpm
+
+RUN mkdir /scripts
+
+COPY *build.sh /scripts
+
+RUN if [[ -f "/scripts/build.sh" ]]; then sh /scripts/build.sh ; fi
```
Comment (joshminnie) [inline services/nginxphp/php54/Dockerfile]: Please add blank line at EOF. Consider configuring your editor to do so on save.

File: services/nginxphp/Readme.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# nginx and Multiple PHP Versions
```
Comment (joshminnie) [inline services/nginxphp/Readme.md]: Change the name of the file to `README.md`

File: services/nginxphp/Readme.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+There are two nginx configuration files that need to be created in order for nginx to function. `default.conf` and `nginx.conf`.
+Both of these files should be placed in the root directory of the service. We have provided `default.conf.example` and
+`nginx.conf.example` that you can copy those files to remove the `.example` from the name so you can configure.
+
+Both of the example files come from a default nginx installation so they should allow nginx to start as they currently are (just
+with no sites configured)
+
+Note: Properly configuring nginx is outside the scope of this document, but here is a hint for setting up a site:
+
+```server {
```
Comment (joshminnie) [inline services/nginxphp/Readme.md]: GitHub Markdown formatting error.

This should be broken out to separate lines. If you have the syntax that you'd like highlighted, that would be on that same line.

File: services/nginxphp/default.conf.example [hunk 4]
```diff
@@ -0,0 +1,44 @@
+server {
```
Comment (joshminnie) [inline services/nginxphp/default.conf.example]: Change file name to `default.sample.conf`

File: services/nginxphp/nginx.conf.example [hunk 5]
```diff
@@ -0,0 +1,31 @@
+user  nginx;
```
Comment (joshminnie) [inline services/nginxphp/nginx.conf.example]: Change file name to `nginx.sample.conf`

File: services/nginxphp/php54/build.sh.example [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (joshminnie) [inline services/nginxphp/php54/build.sh.example]: Change to file name to `build.sample.sh`

File: services/nginxphp/php55/build.sh.example [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (joshminnie) [inline services/nginxphp/php55/build.sh.example]: Change file name to `build.sample.sh`

File: services/nginxphp/php72/build.sh.example [hunk 8]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (joshminnie) [inline services/nginxphp/php72/build.sh.example]: Change file name to `build.sample.sh`

File: services/nginxphp/php72/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,7 @@
+FROM php:7.2-fpm-alpine
+
+RUN mkdir /scripts
+
+COPY *build.sh /scripts
+
+RUN if [[ -f "/scripts/build.sh" ]]; then sh /scripts/build.sh ; fi
```
Comment (joshminnie) [inline services/nginxphp/php72/Dockerfile]: Add blank line at EOF. Consider configuring your editor to handle that automatically for you on save, if you are not already using VS Code.

File: services/nginxphp/php71/build.sh.example [hunk 10]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (joshminnie) [inline services/nginxphp/php71/build.sh.example]: Change file name to `build.sample.sh`

File: services/nginxphp/php71/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,7 @@
+FROM php:7.1-fpm-alpine
+
+RUN mkdir /scripts
+
+COPY *build.sh /scripts
+
+RUN if [[ -f "/scripts/build.sh" ]]; then sh /scripts/build.sh ; fi
```
Comment (joshminnie) [inline services/nginxphp/php71/Dockerfile]: Add blank line at EOF. Consider configuring your editor to handle that automatically for you on save, if you are not already using VS Code.

File: services/nginxphp/php70/build.sh.example [hunk 12]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (joshminnie) [inline services/nginxphp/php70/build.sh.example]: Change file name to `build.sample.sh`

File: services/nginxphp/php70/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,7 @@
+FROM php:7.0-fpm-alpine
+
+RUN mkdir /scripts
+
+COPY *build.sh /scripts
+
+RUN if [[ -f "/scripts/build.sh" ]]; then sh /scripts/build.sh ; fi
```
Comment (joshminnie) [inline services/nginxphp/php70/Dockerfile]: Add blank line at EOF. Consider configuring your editor to handle that automatically for you on save, if you are not already using VS Code.

File: services/nginxphp/php56/build.sh.example [hunk 14]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (joshminnie) [inline services/nginxphp/php56/build.sh.example]: Change file name to `build.sample.sh`.

File: services/nginxphp/php80/build.sh.example [hunk 15]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (joshminnie) [inline services/nginxphp/php80/build.sh.example]: Rename file to `build.sample.sh`

File: services/nginxphp/php74/build.sh.example [hunk 16]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (joshminnie) [inline services/nginxphp/php74/build.sh.example]: Rename file to `build.sample.sh`

**Review by joshminnie — CHANGES_REQUESTED** [review]

**Comment by joshpennington** [comment]
@joshminnie All the requested changes have been completed.

**Review by joshminnie — APPROVED** [review]

**Pull request merged**